lua/nvim-treesitter/locals.lua

```diff
@@ -0,0 +1,80 @@
+-- Functions to handle locals
+-- Locals are a generalization of definition and scopes
+-- its the way nvim-treesitter uses to "understand" the code
+local api = vim.api
+local ts = vim.treesitter
+local queries = require'nvim-treesitter.query'
+local parsers = require'nvim-treesitter.parsers'
+
+local M = {
+  locals={}
+}
+
+function M.is_supported(lang)
+  return queries.get_query(lang, "locals") ~= nil
+end
+
+function M.collect_locals(bufnr)
+  local ft = api.nvim_buf_get_option(bufnr, "ft")
+  local query = queries.get_query(ft, "locals")
+
+  local parser = parsers.get_parser(bufnr)
```

Comment:
not sure if we should bail out here in case the parser is not found.

Reply:
I think that if tests has been done before calling this function (for example in the setup function) we could avoid these cases.

Reply:
i know but in any case, when getting user or environment input, you should always consider every problem that might arise from those inputs, it avoids looking for and fixing bugs later on